tests/scripts/thread-cert/v1_2_test_single_probe.py

```diff
@@ -0,0 +1,103 @@
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS 'AS IS'
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import unittest
+
+import mle
+import thread_cert
+
+LEADER = 1
+SSED_1 = 2
+
+CSL_PERIOD = 500 * 6.25  # 500ms
+CSL_TIMEOUT = 30  # 30s
+
+
+class SSED_SingleProbe(thread_cert.TestCase):
+    TOPOLOGY = {
+        LEADER: {
+            'version': '1.2',
+        },
+        SSED_1: {
+            'version': '1.2',
+            'mode': 's',
+        },
+    }
+    """All nodes are created with default configurations"""
+
+    def test(self):
+
+        self.nodes[SSED_1].set_csl_period(CSL_PERIOD)
+        self.nodes[SSED_1].set_csl_timeout(CSL_TIMEOUT)
+
+        self.nodes[LEADER].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+
+        self.nodes[SSED_1].start()
+        self.simulator.go(7)
+        self.assertEqual(self.nodes[SSED_1].get_state(), 'child')
+
+        leader_addr = self.nodes[LEADER].get_linklocal()
+
+        leader_messages = self.simulator.get_messages_sent_by(LEADER)
+
+        # SSED_1 sends a Single Probe Link Metrics for L2 LQI using MLE Data Request
+        self.nodes[SSED_1].link_metrics_query_single_probe(leader_addr, [0x09])
+        self.simulator.go(5)
+
+        leader_messages = self.simulator.get_messages_sent_by(LEADER)
+        msg = leader_messages.next_mle_message(mle.CommandType.DATA_RESPONSE)
+        msg.assertMleMessageContainsTlv(mle.LinkMetricsReport)
+
+        # SSED_1 sends a Single Probe Link Metrics for Link Margin using MLE Data Request
+        self.nodes[SSED_1].link_metrics_query_single_probe(leader_addr, [0x0a])
+        self.simulator.go(5)
+
+        leader_messages = self.simulator.get_messages_sent_by(LEADER)
+        msg = leader_messages.next_mle_message(mle.CommandType.DATA_RESPONSE)
+        msg.assertMleMessageContainsTlv(mle.LinkMetricsReport)
+
+        # SSED_1 sends a Single Probe Link Metrics for all metrics using MLE Data Request
+        self.nodes[SSED_1].link_metrics_query_single_probe(leader_addr, [0x00, 0x09, 0x0a, 0x0b])
```

Comment:
Argument `typeid_flags` is declared to be bytes, but here it's passed an List[int]. 

Should keep the type consistent ?

Actually I would suggest to just use type `str` and just pass "00090a0b", which seems simplest to type in (and consistent with the CLI command).